assets/common/js/models/GameController.js

```diff
@@ -58,8 +58,8 @@ GameController.prototype.enterWord = function() {
 	let wordToType = this.words.find(word => word.highlightInd > 0) // Find the word that is being typed (is highlighted; highlightInd > 0)
 	let wordToTypeInd = this.words.indexOf(wordToType) // Retrieve index to splice without a second loop
```

Comment:
It seems that you are paying careful attention to avoiding unnecessary loops, which is good. I am not sure if it's due to performance concern or due to focus on functional programming paradigm.

I will add some comments on the time complexity here in case it's the performance that you are concerned about.

While it's true that no loop is used in this block of code, the implementation of the indexOf function itself may very well still loop through the words array. Referencing https://dev.to/lukocastillo/time-complexity-big-0-for-javascript-array-methods-and-examples-mlg shows that the time complexity of the indexOf function is O(n) where n is the number of items in the array, which most likely means that iteration or recursion is involved in the implementation of it. Same can be said for the find function as well as splice.

Since we are constantly removing words that reach the bottom of the screen, I'd be surprised if the number of items in the array gets higher than 30 at any difficulty, so at the moment, I don't believe we will incur too much performance debt even if we liberally use loops for this.words array.